### PR TITLE
Added Support for working with ParticleEffects

### PR DESCRIPTION
**Pull request by zorg93**
Also fixed Entity::Rotation getting and setting using a different rotation order

### CODE REVIEW

**Comment by zorg93** [comment]
What do people think about this setup? I was unsure between making the PTFX Asset start looped effects or make it build a looped effect that can be started and stopped from inside the ParticleEffect class

File: source/core/NativeMemory.cpp [hunk 1]
```diff
@@ -303,6 +303,9 @@ namespace GTA
 			address = FindPattern("\xB2\x01\xE8\x00\x00\x00\x00\x33\xC9\x48\x85\xC0\x74\x3B", "xxx????xxxxxxx");
 			_playerAddressFunc = reinterpret_cast<uintptr_t(*)(int)>(*reinterpret_cast<int *>(address + 3) + address + 7);
 
+			address = FindPattern("\x74\x21\x48\x8B\x48\x20\x48\x85\xC9\x74\x18\x48\x8B\xD6\xE8", "xxxxxxxxxxxxxxx") - 10;
+			_ptfxAddressFunc = reinterpret_cast<uintptr_t(*)(int)>(*reinterpret_cast<int*>(address) + address + 4);
```
Comment (Fireboyd78) [inline source/core/NativeMemory.cpp]: Why is the logic like this?

`(*(int*)address) + address + 4`

So if `address` is 0x100000 which holds a pointer to 0x170000, the result would be 0x270004...this doesn't make sense to me.

Comment (Fireboyd78) [inline source/core/NativeMemory.cpp]: My bad, I just now realized it may be a relative pointer. If that's the case, disregard my comment :P

**Comment by zorg93** [comment]
It's a rip relative offset. So the address is position of offset + offset + 4(end of current instruction, where rip points to during execution of the instruction)

**Pull request merged**